activesupport/lib/active_support/number_helper/rounding_helper.rb

```diff
@@ -0,0 +1,66 @@
+module ActiveSupport
+  module NumberHelper
+    class RoundingHelper # :nodoc:
+      attr_reader :number, :options
+
+      def initialize(number, options)
+        @options = options
+        if precision
+          case number
+          when Float, String
+            @number = BigDecimal(number.to_s)
+          when Rational
+            @number = BigDecimal(number, digit_count(number.to_i) + precision)
+          else
+            @number = number.to_d
+          end
+          @number = round_with_precision
+        else
+          @number = number
+        end
+      end
```

Comment:
This has become quite a complex initializer. I actually think it would be good to keep the *rounding* functionality separate from the initializer (I know I said the opposite in the previous comment – sorry). It’s also confusing that after the initialization, `@number` is a `BigDecimal` if precision is on, but it may be a string or an integer or whatever if precision if off.